app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,41 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = User.find_by(id: params[:user_id])
+    @posts = current_user.posts.includes(:comments)
   end
 
   def show
-    @post = Post.find(params[:id])
+    @user = User.find_by(id: params[:user_id])
+    @post = Post.find_by(id: params[:id])
     @comments = @post.comments
     @likes = @post.likes
+    @new_comment = @post.comments.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+
+    if @post.save
+      @post.update_users_posts_counter
```

Comment:
- [ ] We won't be needing this method anymore since the `after_save` declared in the Post model triggers the method already.